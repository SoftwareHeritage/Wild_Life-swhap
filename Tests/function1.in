A=f1([1])?

A=f1([2])?

A=f1([3])?

A=f2([1|3])?

A=f2([2|4])?

A=f2([1|3])?

A=f2([2|4])?

A=f2([3|X])?

A=f2([X|1])?

A=f3(s(1))?

A=f3(s(2))?

A=f4([123|0])?

A=f4([123|1])?

A=f5([0|123])?

A=f5([1|123])?

A=f6([1,3,5])?

A=f7([s(2),t(3),u(5)])?

A=f7([s(1),t(4),u(6)])?

A=f7([s(1),t(4),u(7)])?

A=f8(s([2|3]))?

A=f8(s([1|4]))?

