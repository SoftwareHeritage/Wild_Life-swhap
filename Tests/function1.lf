
f(A) -> A+123.

f1([{1;2}]) -> 3.

f2([{1;2}|{3;4}]) -> 5.

f3(s({1;2})) -> 6.

f4([S:f(A)|A]) -> S.

f5([A|S:f(A)]) -> S.

f6([{1;2},{3;4},{5;6}]) -> 7.

f7([s(X:{1;2}),t(Y:{3;4}),u(Z:{5;6})]) -> v(X,Y,Z).

f8(s([X:{1;2}|Y:{3;4}])) -> v(X,Y).
