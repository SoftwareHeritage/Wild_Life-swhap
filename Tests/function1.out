> 
*** Yes
> > 
*** Yes
A = 3.
--1> 
*** No
> 
*** Yes
A = 3.
--1> 
*** No
> 
*** No
> > 
*** Yes
A = 5.
--1> 
*** No
> 
*** Yes
A = 5.
--1> 
*** No
> 
*** Yes
A = 5.
--1> 
*** No
> 
*** Yes
A = 5.
--1> 
*** No
> 
*** No
> > 
*** No
> > 
*** Yes
A = 6.
--1> 
*** No
> 
*** Yes
A = 6.
--1> 
*** No
> 
*** Yes
A = 123.
--1> 
*** No
> 
*** Yes
A = 123.
--1> 
*** No
> 
*** Yes
A = 123.
--1> 
*** No
> 
*** No
> > 
*** Yes
A = 7.
--1> 
*** No
> 
*** Yes
A = v(2,3,5).
--1> 
*** No
> 
*** Yes
A = v(1,4,6).
--1> 
*** No
> 
*** No
> > 
*** Yes
A = v(2,3).
--1> 
*** No
> 
*** Yes
A = v(1,4).
--1> 
*** No
> 
